### ISSUE TITLE
linux: NewKey panics after last key removed

### ISSUE THREAD

**Issue by wadey** [facebookincubator/sks#3]
This is happening with commit d4f546b92498d6dd1c6307d1fab79d110d698bde

I can run the example in the README once, which creates a NewKey and then Removes it at the end. The next time I try to run it, I get a panic from NewKey.

The panic is here: https://github.com/facebookincubator/sks/blob/d4f546b92498d6dd1c6307d1fab79d110d698bde/linux/handler.go#L153-L160

It looks like all of the handles get filtered out by `handlesToSlice` and nothing checks for that case afterwards.

```
V0710 10:10:20.879136   75101 tpm.go:160] Initialized key handler
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/facebookincubator/sks/linux.(*KeyHandler).nextAvailable(0xc000190490)
	/go/pkg/mod/github.com/facebookincubator/sks@v0.0.0-20230710110558-d4f546b92498/linux/handler.go:160 +0x373
github.com/facebookincubator/sks/linux.(*KeyHandler).Get(0xc000190490, {0x5eb5e4, 0x14})
	/go/pkg/mod/github.com/facebookincubator/sks@v0.0.0-20230710110558-d4f546b92498/linux/handler.go:71 +0x15b
github.com/facebookincubator/sks/linux.(*tpmDevice).GenKeyPair(0xc000190480, {0x5eb5e4, 0x14})
	/go/pkg/mod/github.com/facebookincubator/sks@v0.0.0-20230710110558-d4f546b92498/linux/sks.go:38 +0x85
github.com/facebookincubator/sks.genKeyPair({0x5eb5e4, 0x14}, {0x5eb5f9, 0x5}, 0x1?, 0x0?)
	/go/pkg/mod/github.com/facebookincubator/sks@v0.0.0-20230710110558-d4f546b92498/sks_linux.go:51 +0x1da
github.com/facebookincubator/sks.NewKey({0x5eb5e4, 0x14}, {0x5eb5f9, 0x5}, 0xa0?, 0x61?, {0x0?, 0x46261b?, 0xc0000061a0?})
	/go/pkg/mod/github.com/facebookincubator/sks@v0.0.0-20230710110558-d4f546b92498/key.go:88 +0x12c
```

**Comment by shapzka** [comment]
Hey @wadey - we are looking into this. Thanks for raising the issue!

**Comment by u1f35c** [comment]
Resolved by 6cd29d7; on a fresh system we ended up with a persistent key for the root that wasn't in the handle allocation range, so the earlier check for no handles didn't fire, but we filtered out the root. Check + test added.
